tests/test_army_firearms.py

```
@@ -92,71 +99,114 @@ def test_create_army_with_firearm(game):
+    detail_hi = db.army_detail(name)
+    assert detail != detail_hi
+    assert "91" in detail_hi and "7" in detail_hi
+    assert "91" not in detail
 
-def test_army_report_shows_firearm_and_cannon(read_game):
-    """army_report(list_armies 警讯)带火器 + 随军大炮(炮)，read 摘要面闭环（CMR codexC）。"""
-    db, _, _ = read_game
-    rpt = db.army_report(limit=8)
-    assert "火器" in rpt
-    assert "炮" in rpt
+    # report 消费两轴：炮门可数；火器变化使出口可判别
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)